### PR TITLE
Rebase to bash-5.0

### PR DESCRIPTION
**Pull request by siteshwar**

### CODE REVIEW

File: bash.spec [hunk 1]
```diff
@@ -21,80 +21,64 @@ Source3: dot-bash_logout
...
-# 1556867 - case in a for loop inside subshell causes syntax error
-# This should be dropped while rebasing to bash-4.5
-Patch131: bash-4.4-case-in-command-subst.patch
-
-# 1637018
-Patch132: bash-4.4-coverity.patch
+# Patch101: bash-2.02-security.patch -- It was taken from bash-2.0 and I can't see a reason why it should be kept.
+# Patch102: bash-2.03-paths.patch -- We don't want to add '/etc:/usr/etc' in standard utils path, so it should be kept.
+# Patch103: bash-2.03-profile.patch -- It should be kept if we want to retain this feature for backward compatiblity.
+# Patch104: bash-2.05a-interpreter.patch --  No idea about it.
```
Comment (kdudka) [inline bash.spec]: Likely non-upstreamable and buggy.  See https://bugzilla.redhat.com/show_bug.cgi?id=60870#c4 .

File: bash.spec [hunk 2]
```diff
@@ -21,80 +21,64 @@ Source3: dot-bash_logout
...
+# Patch105: bash-2.05b-debuginfo.patch -- It should be kept to generate info for debuginfo files.
+# Patch106: bash-2.05b-manso.patch -- I am not sure what this patch solves.
```
Comment (kdudka) [inline bash.spec]: Safe to drop IMO.

File: bash.spec [hunk 3]
```diff
@@ -21,80 +21,64 @@ Source3: dot-bash_logout
...
+# Patch107: bash-2.05b-pgrp_sync.patch -- Keep it.
+# Patch108: bash-2.05b-xcc.patch -- This change is already in upstream. This patch should be removed.
+# Patch109: bash-3.2-audit.patch -- This enables audit logs. Keep it.
+# Patch110: bash-3.2-ssh_source_bash.patch -- It sources bashrc files when run under ssh. Keep it.
+# Patch112: bash-infotags.patch -- Keep it.
+# Patch113: bash-requires.patch -- Keep it.
+# Patch114: bash-setlocale.patch -- Keep it.
```
Comment (kdudka) [inline bash.spec]: Bugzilla says this was reported upstream: https://bugzilla.redhat.com/show_bug.cgi?id=165249#c3  If the bug still exists in the latest upstream release, it should be reported again.  I see no reason for keeping this patch downstream.

File: bash.spec [hunk 4]
```diff
@@ -21,80 +21,64 @@ Source3: dot-bash_logout
...
+# Patch115: bash-tty-tests.patch -- These tests will fail under mock build, so disable them.
```
Comment (kdudka) [inline bash.spec]: We should IMO check that it is still the case.  I was able to successfully remove a similar patch from zsh recently.

File: bash.spec [hunk 5]
```diff
@@ -21,80 +21,64 @@ Source3: dot-bash_logout
...
+# 
+# # 484809, check if interp section is NOBITS
+# Patch116: bash-4.0-nobits.patch -- Keep it.
```
Comment (kdudka) [inline bash.spec]: This changes the code introduced by bash-2.05a-interpreter.patch and also breaks indentation at some places.  We should merge the patches together (perhaps as 2 git commits serialized by `git format-patch`) in a single file and fix the broken indentation.

File: bash.spec [hunk 6]
```diff
@@ -21,80 +21,64 @@ Source3: dot-bash_logout
...
+# 
+# # Do the same CFLAGS in generated Makefile in examples
+# Patch117: bash-4.1-examples.patch -- Keep it ?
```
Comment (kdudka) [inline bash.spec]: Please insert a comment that this patch was added to resolve a multilib conflict on RPM packages.

File: bash.spec [hunk 7]
```diff
@@ -21,80 +21,64 @@ Source3: dot-bash_logout
...
+# 
+# # Builtins like echo and printf won't report errors
+# # when output does not succeed due to EPIPE
+# Patch118: bash-4.1-broken_pipe.patch -- Keep it.
+# 
+# # Enable system-wide .bash_logout for login shells
+# Patch119: bash-4.2-rc2-logout.patch -- Keep it.
+# 
+# # Static analyzis shows some issues in bash-2.05a-interpreter.patch
+# Patch120: bash-4.2-coverity.patch -- It should be kept if we keep bash-2.05a-interpreter.patch.
```
Comment (kdudka) [inline bash.spec]: I would also merge this one so that all ELF-handling patch are at a single place.

File: bash.spec [hunk 8]
```diff
@@ -21,80 +21,64 @@ Source3: dot-bash_logout
...
+# 
+# # Don't call malloc in signal handler
+# Patch121: bash-4.1-defer-sigchld-trap.patch -- This patch seems incomplete, remove it ? Chet said that it should be fixed already https://lists.gnu.org/archive/html/bug-bash/2012-03/msg00045.html
```
Comment (kdudka) [inline bash.spec]: Yes, this patch does nothing as I understand it.

File: bash.spec [hunk 9]
```diff
@@ -21,80 +21,64 @@ Source3: dot-bash_logout
...
+# 
+# # 799958, updated info about trap
+# Patch122: bash-4.2-manpage_trap.patch -- Keep this patch and try to upstream it ?
```
Comment (kdudka) [inline bash.spec]: Yes, looks like either bash packager forgot to send it upstream, or upstream forgot to commit it.

File: bash.spec [hunk 10]
```diff
@@ -21,80 +21,64 @@ Source3: dot-bash_logout
...
+# 
+# #1241533,1224855 - bash leaks memory when LC_ALL set
+# Patch126: bash-4.3-memleak-lc_all.patch -- Keep it and upstream it.
+# 
+# # bash-4.4 builds loadable builtin examples by default
+# # this patch disables it
+# Patch127: bash-4.4-no-loadable-builtins.patch -- Keep it.
+# 
+# # 1637018
+# Patch132: bash-4.4-coverity.patch -- Keep it and upstream it.
```
Comment (kdudka) [inline bash.spec]: I believe this was already sent upstream.  Is there still something that we want to keep and upstream has not fixed yet?

Comment (siteshwar) [reply bash.spec]: I sent it upstream, but I need to verify if it was merged before bash-5.0 release.

**Pull request closed without merging**